overlay-ot.conf

```diff
@@ -0,0 +1,43 @@
+# Hack: Disable BT on devices which support both for now
+CONFIG_BT=n
+
+# Enable C++
+CONFIG_CPLUSPLUS=y
+
+# Enable flash page layout extension
+CONFIG_FLASH_PAGE_LAYOUT=y
+
+# Disable Proxy
+CONFIG_LWM2M_FIRMWARE_UPDATE_PULL_COAP_PROXY_SUPPORT=n
+
+# Disable IPv4
+CONFIG_NET_IPV4=n
+CONFIG_NET_CONFIG_NEED_IPV4=n
+CONFIG_NET_CONFIG_MY_IPV4_ADDR=""
+CONFIG_NET_CONFIG_PEER_IPV4_ADDR=""
+
+# OpenThread doesn't use NS/NA
+CONFIG_NET_IPV6_NBR_CACHE=n
+CONFIG_NET_IPV6_MLD=n
+
+# OpenThread requires larger stack sizes
+CONFIG_MAIN_STACK_SIZE=2048
+CONFIG_SYSTEM_WORKQUEUE_STACK_SIZE=2048
+CONFIG_NET_TX_STACK_SIZE=2048
+CONFIG_NET_RX_STACK_SIZE=5120
+
+# Disable conflicting IEEE802154 L2
+CONFIG_NET_L2_IEEE802154=n
+CONFIG_NET_L2_IEEE802154_SHELL=n
+
+# Enable OpenThread
+CONFIG_NET_L2_OPENTHREAD=y
+
+# OpenThread will set local IP
+CONFIG_NET_CONFIG_MY_IPV6_ADDR=""
```

Comment:
Why is this needed? The option has no defaults, so it implicitly defaults to the empty string.